packages/multichain/src/caip25Permission.ts

```diff
@@ -0,0 +1,283 @@
+      }
+
+      if (
+        !isEqual(requiredScopes, normalizedRequiredScopes) ||
+        !isEqual(optionalScopes, normalizedOptionalScopes)
+      ) {
+        throw new Error(
+          `${Caip25EndowmentPermissionName} error: Received non-normalized value for caveat of type "${Caip25CaveatType}".`,
+        );
+      }
+    },
+  };
+};
+
+/**
+ * The `caip25` endowment specification builder. Passed to the
+ * `PermissionController` for constructing and validating the
+ * `endowment:caip25` permission.
+ */
+export const caip25EndowmentBuilder = Object.freeze({
+  targetName: Caip25EndowmentPermissionName,
+  specificationBuilder,
+} as const);
+
+/**
+ * Factories that construct caveat mutator functions that are passed to
+ * PermissionController.updatePermissionsByCaveat.
+ */
+export const Caip25CaveatMutators = {
+  [Caip25CaveatType]: {
+    removeScope,
+    removeAccount,
+  },
+};
+
+/**
+ * Removes the account from the scope object.
+ *
+ * @param targetAddress - The address to remove from the scope object.
+ * @returns A function that removes the account from the scope object.
+ */
+function removeAccountFilterFn(targetAddress: string) {
+  return (account: CaipAccountId) => {
+    const parsed = parseCaipAccountId(account);
+    return parsed.address !== targetAddress;
+  };
+}
+
+/**
+ * Removes the account from the scope object.
+ *
+ * @param scopeObject - The scope object to remove the account from.
+ * @param targetAddress - The address to remove from the scope object.
+ */
+function removeAccountFromScopeObject(
+  scopeObject: InternalScopeObject,
+  targetAddress: string,
+) {
+  if (scopeObject.accounts) {
+    scopeObject.accounts = scopeObject.accounts.filter(
+      removeAccountFilterFn(targetAddress),
+    );
+  }
+}
+
+/**
+ * Removes the target account from the scope object.
+ *
+ * @param caip25CaveatValue - The CAIP-25 permission caveat value from which to remove the account (across all chain scopes).
+ * @param targetAddress - The address to remove from the scope object. Not a CAIP-10 formatted address because it will be removed across each chain scope.
+ * @returns The updated scope object.
+ */
+function removeAccount(
+  caip25CaveatValue: Caip25CaveatValue,
+  targetAddress: string,
+) {
+  const copyOfCaveatValue = cloneDeep(caip25CaveatValue);
+
+  [copyOfCaveatValue.requiredScopes, copyOfCaveatValue.optionalScopes].forEach(
+    (scopes) => {
+      Object.entries(scopes).forEach(([, scopeObject]) => {
+        removeAccountFromScopeObject(scopeObject, targetAddress);
+      });
+    },
+  );
+
+  const noChange = isEqual(copyOfCaveatValue, caip25CaveatValue);
+
+  if (noChange) {
+    return {
+      operation: CaveatMutatorOperation.Noop,
+    };
+  }
+
+  return {
+    operation: CaveatMutatorOperation.UpdateValue,
+    value: copyOfCaveatValue,
+  };
+}
+
+/**
+ * Removes the target account from the value arrays of all
+ * `endowment:caip25` caveats. No-ops if the target scopeString is not in
```

Comment:
Isn't it just a single caveat? 🤔 

```suggestion
 * Removes the target account from the value arrays of the given
 * `endowment:caip25` caveat. No-ops if the target scopeString is not in
```